### PR TITLE
Fix projectmodel-server regression

### PR DESCRIPTION
**Pull request by troydai**
1. Address the null reference exception when a project dependency is
broken.
2. Address the duplicate key issues after the logic of redirecting
System pacage to reference assembly was added
3. Include projectmodel-server test in the build script

Issue: #1098 
/cc @davidfowl

### CODE REVIEW

File: src/Microsoft.DotNet.ProjectModel/Compilation/LibraryExporter.cs [hunk 1]
```diff
@@ -160,6 +160,12 @@ private LibraryExport ExportProject(ProjectDescription project)
             var runtimeAssets = new List<string>();
             var sourceReferences = new List<string>();
 
+            if (project.Project == null)
```
Comment (davidfowl) [inline src/Microsoft.DotNet.ProjectModel/Compilation/LibraryExporter.cs]: Why not just handle unresolved projects generally

File: src/Microsoft.DotNet.ProjectModel/ProjectContextBuilder.cs [hunk 2]
```diff
@@ -292,43 +293,56 @@ public ProjectContext Build()
                         requiresFrameworkAssemblies = true;
 
                         // Remove the original package reference
+                        var originalReference = libraries[pair.Key];
```
Comment (davidfowl) [inline src/Microsoft.DotNet.ProjectModel/ProjectContextBuilder.cs]: This is unused

File: src/Microsoft.DotNet.ProjectModel/ProjectContextBuilder.cs [hunk 3]
```diff
@@ -292,43 +293,56 @@ public ProjectContext Build()
...
+                                                    UnresolvedDependencyProvider.GetDescription(dependency, TargetFramework);
+                            libraries[key] = dependencyDescription;
                         }
                         else
                         {
-                            dep = UnresolvedDependencyProvider.GetDescription(dependency, TargetFramework);
-                            libraries[key] = dep;
+                            // a dependency which type is unspecified fails to match, then try to find a 
+                            // reference assembly type dependency
+                            if (!libraries.TryGetValue(new LibraryKey(dependency.Name, LibraryType.ReferenceAssembly), out dependencyDescription))
```
Comment (davidfowl) [inline src/Microsoft.DotNet.ProjectModel/ProjectContextBuilder.cs]: Feels like this should be an else if

**Comment by troydai** [comment]
@dotnet-bot test this please

**Comment by troydai** [comment]
@dotnet-bot test on Ubuntu Debug Build please

**Comment by troydai** [comment]
@dotnet-bot test this please

**Comment by troydai** [comment]
@dotnet-bot test this please.

**Comment by troydai** [comment]
@mmitche could you please help me take a look the `Windows_NT Release Build`? It seems to me that some process is running and prevent it from being cleaned.

**Comment by troydai** [comment]
Ping @davidfowl

File: testapp/DthTestProjects/src/IncompatiblePackageSample/project.json [hunk 4]
```diff
@@ -1,6 +1,7 @@
 {
     "dependencies": {
-        "Newtonsoft.Json": "4.5.11"
+        "NETStandard.Library": "1.0.0-rc2-23727",
```
Comment (davidfowl) [inline testapp/DthTestProjects/src/IncompatiblePackageSample/project.json]: Change to match the new version.

Comment (troydai) [inline testapp/DthTestProjects/src/IncompatiblePackageSample/project.json]: Will update to 23728

**Comment by troydai** [comment]
@dotnet-bot test Ubuntu Debug Build please

**Comment by troydai** [comment]
@dotnet-bot test Ubuntu Debug Build please

**Comment by troydai** [comment]
/cc @davidfowl @piotrpMSFT @Sridhar-MS 
The build on __Ubuntu Debug Build__ keep failing after restore. It is not related to the change in this PR. I'd like to move on to merge the change. Any concerns?

**Comment by troydai** [comment]
@dotnet-bot test Ubuntu Debug Build please

**Comment by troydai** [comment]
@dotnet-bot test Ubuntu Debug Build please

**Comment by troydai** [comment]
/cc @davidfowl @piotrpMSFT 

i’m going to merge this to `rel/1.0.0` despite the build failure on __Ubuntu Debug Build__. The failure is caused by restore failure long before any code related to this change is compiled.

**Pull request merged**